Automattic/wp-calypso
# Delete Site: fix delete validation

Related to https://github.com/Automattic/dotcom-forge/issues/5070

## Proposed Changes
Fixes confirm site deletion modal, where the letter case is ignored:

| **Before** | **After** |
| :---:         |    :---:       |
|![2024-01-09T15:45:25,291069015+02:00](https://github.com/Automattic/wp-calypso/assets/497103/e70da252-1ef2-49f5-b514-83356341f443)|![2024-01-09T15:49:43,956860245+02:00](https://github.com/Automattic/wp-calypso/assets/497103/0e095c97-e87c-4fd3-9487-6565b3539f81)|

## Testing Instructions


- Create a free site
- Navigate to `Settings > General`
- Navigate to the bottom of the page and click the `Delete your site permanently`
![2024-01-09T16:00:54,281226537+02:00](https://github.com/Automattic/wp-calypso/assets/497103/2b79d81d-da60-4641-9be8-dc88644cf82c)
- Press the `Delete site` button.
- In the confirmation dialog write your site slug but turn some letters to uppercase.
- Ensure that you are not able to press the `Delete this site` button`.

## Pre-merge Checklist
